internal/cmd/flink/command_shell.go

```diff
@@ -32,18 +34,40 @@ func (c *command) newShellCommand(prerunner pcmd.PreRunner) *cobra.Command {
 	return cmd
 }
 
-func (c *command) authenticated(authenticated func(*cobra.Command, []string) error, cmd *cobra.Command) func() error {
+func (c *command) authenticated(authenticated func(*cobra.Command, []string) error, cmd *cobra.Command, jwtValidator pcmd.JWTValidator) func() error {
 	return func() error {
 		cfg, err := load.LoadAndMigrate(v1.New())
 		if err != nil {
 			return err
 		}
-		auth := cfg.Context().State.AuthToken
+
+		authToken := cfg.Context().State.AuthToken
 		authRefreshToken := cfg.Context().State.AuthRefreshToken
-		if err := c.Context.UpdateAuthTokens(auth, authRefreshToken); err != nil {
+		if err := c.Context.UpdateAuthTokens(authToken, authRefreshToken); err != nil {
+			return err
+		}
+
+		if err := authenticated(cmd, nil); err != nil {
+			return err
+		}
+
+		flinkGatewayClient, err := c.GetFlinkGatewayClient()
+		if err != nil {
 			return err
 		}
-		return authenticated(cmd, nil)
+
+		jwtCtx := &v1.Context{State: &v1.ContextState{AuthToken: flinkGatewayClient.AuthToken}}
+		if tokenErr := jwtValidator.Validate(jwtCtx); tokenErr != nil {
+			output.Println("Token expired. Refreshing token...")
```

Comment:
is this something we want to output to customer?

Reply:
I thought it could be helpful, but could also delete this. But thanks to your comment I noticed another thing: we would be printing the error twice. Once in line 64 and then again in input_controller line 84!